src/main/java/org/jenkinsci/plugins/workflow/cps/nodes/StepAtomNode.java

```diff
@@ -48,7 +48,7 @@
 
     public StepAtomNode(CpsFlowExecution exec, StepDescriptor d, FlowNode parent) {
         super(exec, exec.iotaStr(), parent);
-        this.descriptorId = d!=null ? d.getId() : null;
+        this.descriptorId = d!=null ? d.getId().intern() : null;
```

Comment:
I'm pretty sure that is *not* being cached, because I definitely see dupes in memory when looking with YourKit and they're gone after the change